test/integration/rules/aria-form-field-label/aria-form-field-label.html

```diff
@@ -0,0 +1,176 @@
+<!-- PASS -->
+<!-- checkbox -->
+<div id="pass1" aria-label="newspaper" role="checkbox">Newspaper</div>
+<!-- combobox -->
+<div id="pass2" aria-label="country" role="combobox">England</div>
+<!-- listbox -->
+<p id="pass3Label" role="label">Select a color:</p>
+<div id="pass3" role="listbox" aria-labelledby="pass3Label">
+	<div role="option">Orange</div>
+</div>
+<!-- menuitemchecbox -->
+<ul role="menu">
+	<li role="menuitem">Console</li>
+	<li
+		id="pass4"
+		role="menuitemcheckbox"
+		aria-label="Word wrap"
+		aria-checked="true"
+	>
+		Word wrap
+	</li>
+</ul>
+<!-- menuitemradio -->
+<ul role="menu">
+	<li
+		id="pass5"
+		role="menuitemradio"
+		aria-label="Sans-serif"
+		aria-checked="true"
+	>
+		Sans-serif
+	</li>
+</ul>
+<!-- radio -->
+<div role="radiogroup">
+	<div
+		id="pass6"
+		role="radio"
+		aria-checked="false"
+		tabindex="0"
+		aria-label="Regular crust"
+	>
+		Regular crust
+	</div>
+</div>
+<!-- searchbox -->
+<p id="pass7Label" role="label">Search currency pairs:</p>
+<div
+	id="pass7"
+	role="searchbox"
+	contenteditable="true"
+	aria-labelledby="pass7Label"
+></div>
+<!-- slider -->
+<div
+	id="pass8"
+	role="slider"
+	aria-label="Choose a value"
+	aria-valuemin="1"
+	aria-valuemax="7"
+	aria-valuenow="2"
+></div>
+<!-- spinbutton -->
+<div
+	id="pass9"
+	role="spinbutton"
+	aria-valuemin="0"
+	aria-valuemax="10"
+	aria-valuenow="8"
+	aria-label="Enter quantity:"
+></div>
+<!-- switch -->
+<div
+	id="pass10"
+	role="switch"
+	aria-checked="true"
+	aria-label="Display blue light:"
+>
+	<span>off</span>
+	<span>on</span>
+</div>
+<!-- textbox -->
+<label id="foo">
+	foo
+	<div id="pass11" role="textbox" aria-labelledby="foo"></div>
+</label>
+
+<!-- FAIL -->
+<!-- aria-label with empty text string -->
+<div id="fail1" aria-label=" " role="combobox">England</div>
+<!-- The label does not exist. -->
+<div id="fail2" aria-labelledby="non-existing" role="combobox">England</div>
+<!-- The implicit label is not supported on div elements. -->
+<label>
+	first name
+	<div id="fail3" role="textbox"></div>
+</label>
+<!-- explicit label -->
+<label for="fail4">first name</label>
+<div role="textbox" id="fail4"></div>
+<!-- checkbox -->
+<div id="fail5" role="checkbox">Newspaper</div>
+<!-- combobox -->
+<div id="fail6" role="combobox">England</div>
+<!-- listbox -->
+<div id="fail7" role="listbox" aria-labelledby="label-does-not-exist">
+	<div role="option">Orange</div>
+</div>
+<!-- menuitemchecbox -->
+<ul role="menu">
+	<li role="menuitem">Console</li>
+	<li id="fail8" role="menuitemcheckbox" aria-checked="true">Word wrap</li>
```

Comment:
Darnit... I think we made a mistake Jey. Some of these roles are named from contents, and others aren't. This role should pass, because "word wrap" is its accessible name. We may have to break up this rule into two. One for named from author only, and one for named from contents. Here's the breakdown:

### Named from Contents
checkbox
menuitemcheckbox
menuitemradio
radio
switch

### Named by author only
combobox
listbox
searchbox
slider
spinbutton
textbox

Reply:
@WilcoFiers 

As discussed the other Rule (`aria-input-field-label`) PR is here - https://github.com/dequelabs/axe-core/pull/1610, that handles `named by author only`.